core/modules/book/book.module

```diff
@@ -1283,3 +1292,77 @@ function book_menu_subtree_data($link) {
 
   return $tree[$cid];
 }
+
+/**
+ * Invalidates cached data relating to book_cache.
+ *
+ * @param int|null $cid
+ *   (optional) Cache ID of the record to clear. If empty, all records will be
+ *   cleared from the table. Defaults to NULL.
+ * @param bool $wildcard
+ *   (optional) If TRUE, cache IDs starting with $cid are deleted in addition to
+ *   the exact cache ID specified by $cid. Defaults to FALSE.
+ */
+function _book_cache_clear($cid = NULL, $wildcard = FALSE) {
+  if (empty($cid)) {
+    cache_clear_all('*', 'cache_book', TRUE);
+  }
+  else {
+    cache_clear_all($cid, 'cache_book', $wildcard);
+  }
+}
+
+/**
+ * Implements hook_flush_caches().
+ */
+function book_flush_caches() {
+  _book_cache_clear();
+  return array();
+}
+
+/**
+ * Implements hook_admin_menu_cache_info().
```

Comment:
You probably meant `hook_admin_bar_cache_info()`. 

The hook `hook_admin_menu_cache_info()` doesn't exist in Backdrop